akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -370,14 +372,17 @@ private[stream] final class VirtualProcessor[T] extends Processor[T, T] {
           try {
             subscriptionStatus.set(s)
             tryOnSubscribe(s, sub)
+            if (isCancelled) throw canNotSubscribeTheSameSubscriberMultipleTimesException
             sub.closeLatch() // allow onNext only now
             terminationStatus.getAndSet(Allowed) match {
               case null       ⇒ // nothing happened yet
               case Completed  ⇒ tryOnComplete(s)
               case Failed(ex) ⇒ tryOnError(s, ex)
               case Allowed    ⇒ // all good
+
             }
           } catch {
+            case ex @ canNotSubscribeTheSameSubscriberMultipleTimesException ⇒ throw ex
```

Comment:
Yeah, my bad. 
Code snippet make sense as tryComplete and tryError do not cancel VirtualProcessor in the way onSubscription does